horace_core/sqw/@sqw/private/split_npix_full_bins.m

```diff
@@ -0,0 +1,83 @@
+function [npix_chunks, idxs] = split_npix_full_bins(npix, page_size, num_pix, npix_cum_sum)
+%SPLIT_NPIX_FULL_BINS Split the given array of counts such that the sum of each
+%  sub-array has a maximum of page_size.
+%
+% The npix counts are not split across bin boundaries - i.e. each sub-array
+% contains only full bins. If a value in npix is greater than page_size, then
+% that value will comprise its own sub-array.
+%
+% Input:
+% ------
+% npix          An array of integers (usually sqw.data.npix)
+% page_size     An integer specifying the maximum sum for each sub-array
+% num_pix       The sum of npix [optional].
+%               If this argument is not given it is calculated on-the-fly.
+%               In most cases, this value is cached for sqw objects in
+%               sqw.data.pix.num_pixels.
+% npix_cum_sum  The cumulative sum of the given npix array [optional].
+%               If not specified it will be calculated on-the-fly.
+%               Given as an optional argument as it's common to have already
+%               calculated this quantity (defines upper bounds of bin edges).
+%               This should be a vector.
+%
+% Output:
+% -------
+% npix_chunks  Cell array of vectors. Concatenation of these arrays will be
+%              equal to a flattened version the inputted npix array, i.e.
+%              npix(:).
+% idxs         The indices at which the input npix array was "split". Has size
+%              [2, n], where n is numel(npix_chunks). Each idxs(i, :) are the
```

Comment:
Ordering is important here is `(1, n)` the upper or lower index?

The description looks to be transposed from where it is used above: there `sub_bin_idxs(1, iter):sub_bin_idxs(2, iter)`
so `Each idxs(:, i) are the lower and upper indices ...`

Reply:
Ah, well spotted, cheers